src/serac/physics/utilities/functional/array.hpp

```diff
@@ -0,0 +1,376 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
```

Comment:
In the quadrature data GPU branch, we're using chai array. Should we use that here? Is the ultimate goal axom array? Thoughts @joshessman-llnl ?